src/Stratis.Bitcoin/Utilities/AsyncQueue.cs

```diff
@@ -107,6 +107,20 @@ public void Enqueue(T item)
             }
         }
 
+        /// <summary>
+        /// The number of items int he queue.
+        /// </summary>
+        public int Count
```

Comment:
I remember @Aprogiena was against adding `Count` to AQ but he didn't mind inheriting from AQ and extending it. 
So probably better to do it like that and use extended AQ where count is needed.

Reply:
Reason is that Count can be misused easily.